JSUserRegistrationService/src/test/java/com/example/jsuserregistrationservice/UserTest.java

```diff
@@ -0,0 +1,274 @@
+package com.example.jsuserregistrationservice;
+
+import static org.junit.Assert.*;
+
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestContextManager;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+/**
+ * Copyright (c) 2016, 2017, Jumia.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class UserTest {
+
+    @Autowired
+    private LocalValidatorFactoryBean validator;
+
+    private TestContextManager testContextManager;
+
+    @Test
+    public void testFirstNameIsValid() throws Exception {
+        // Valid Scenario
+        User user = new User();
+        PersonalDetails userPersonalDetails = new PersonalDetails();
+
+        userPersonalDetails.setFirstName("Ali");
+        userPersonalDetails.setLastName("Dawood");
+        userPersonalDetails.setAge(32);
+
+        user.setPersonalDetails(userPersonalDetails);
+
+        Set<ConstraintViolation<PersonalDetails>> result = this.validator.validateProperty(userPersonalDetails,"firstName");
+
+        assertTrue(result.isEmpty());
+
+        //Invalid Scenario firstName is not set
```

Comment:
For readability and debug purposes, each scenario should be contained in its own Test method.